app/models/podcast_planner.rb

```diff
@@ -0,0 +1,152 @@
+class PodcastPlanner
+  attr_accessor :podcast_id, :dates, :selected_days, :week_condition, :period, :monthly_weeks, :start_date, :date_range_condition, :number_of_episodes, :end_date, :publish_time, :segment_count, :drafts
+
+  def initialize(params = {})
+    @dates = []
+    @drafts = []
+    @podcast_id = params[:podcast_id]
+    @start_date = params[:start_date].try(:to_datetime)
+    @selected_days = params[:selected_days].try { reject(&:blank?) }.try { map { |day| day.to_i } }
+    @period = params[:period].try(:to_i)
+    @monthly_weeks = params[:monthly_weeks].try { reject(&:blank?) }.try { map { |week| week.to_i } }
+    @date_range_condition = params[:date_range_condition]
+    @week_condition = params[:week_condition]
+    @number_of_episodes = params[:number_of_episodes].try(:to_i)
+    @end_date = params[:end_date].try(:to_datetime)
+    @publish_time = params[:publish_time].try(:to_time)
+    @segment_count = params[:segment_count].try(:to_i)
+  end
+
+  def date_range_ends_by_episodes?
+    @date_range_condition == "episodes"
+  end
+
+  def date_range_ends_by_end_date?
+    @date_range_condition == "date"
+  end
+
+  def episodes_remain?
+    @dates.length < @number_of_episodes
+  end
+
+  def end_date_passed?(date)
+    date > @end_date
+  end
+
+  def periodic?
+    @week_condition == "periodic"
+  end
+
+  def monthly?
+    @week_condition == "monthly"
+  end
+
+  def ready_to_select_weeks?
```

Comment:
this method and the next two below are what i'm using as guards when generating dates. i'm wondering if i should just put these into actual validations, or if it makes sense to place the logic here? it feels like i won't need it other than as a guard, which feels like a hint to put it into a validation, but i'm not sure on which(or either/neither) is best practice.